src/contracts/core/DelegationManager.sol

```diff
@@ -264,8 +278,38 @@ contract DelegationManager is Initializable, OwnableUpgradeable, Pausable, Deleg
         // actually undelegate the staker
         emit StakerUndelegated(staker, operator);
         delegatedTo[staker] = address(0);
+    }
 
-        return withdrawalRoot;
+    /**
+     * @notice Called by a staker to exit the "undelegation limbo" mode.
+     * @param middlewareTimesIndex Passed on as an input to the `slasher.canWithdraw` function, to ensure that the caller can exit undelegation limbo.
+     * This is because undelegation limbo is subject to some of the same restrictions as completing a withdrawal, to ensure that a staker cannot use undelegation
+     * limbo to avoid potentially being subject to slashing.
+     * @dev Note that this checks against the frozen status of the operator *that the caller was delegated to when they entered undelegation limbo*, to provide
+     * the same kind of slashing eligibility as withdrawals via the StrategyManager or DelegationManager.
+     */
+    function exitUndelegationLimbo(
+        uint256 middlewareTimesIndex
+    ) external onlyNotFrozen(_stakerUndelegationLimboStatus[msg.sender].delegatedAddress) {
+        require(
+            isInUndelegationLimbo(msg.sender),
+            "DelegationManager.exitUndelegationLimbo: must be in undelegation limbo"
+        );
+
+        require(
+            slasher.canWithdraw(
+                _stakerUndelegationLimboStatus[msg.sender].delegatedAddress,
+                _stakerUndelegationLimboStatus[msg.sender].startBlock,
+                middlewareTimesIndex
+            ),
+            "DelegationManager.exitUndelegationLimbo: shares in limbo are still slashable"
+        );
+
+        // a staker cannot be simultaneously in delegation limbo and delegated to an operator, so we do not need to return any shares to the delegation system here
+
+        // delete the pod owner's undelegation limbo details and emit an event
+        delete _stakerUndelegationLimboStatus[msg.sender];
+        emit UndelegationLimboExited(msg.sender);
```

Comment:
I'm not sure I follow the logic behind not enforcing a delay here. Seems like this allows you to very quickly swap out your Operator if you think they're about to get slashed or something.

Reply:
discussed this some offline. the intention is that
- the minimum withdrawal delay is an extra safety measure on funds leaving EigenLayer, not designed to have anything to do with ensuring slashing
- the call to `slasher.canWithdraw` should enforce slashing eligibility, when slashing is actually enabled